### PR TITLE
Feature/routes websocket signal

### PR DESCRIPTION
**Pull request by JPHutchins**
Implement and test byte stream management.

### CODE REVIEW

File: backend/gatterserver/streams/__init__.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+        def _send_wrapper(stream_id: StreamId) -> Awaitable:
+            async def _send(data: bytes):
+                async with self._lock:
+                    try:
+                        self._pending_data.append(StreamPacket(stream_id, data))
+                    except Exception as e:
+                        LOGGER.critical(e, exc_info=True)
+                    finally:
+                        self._semaphore.release()
```
Comment (JPHutchins) [inline backend/gatterserver/streams/__init__.py]: Actually if adding the data to the queue fails we would not want to increment the semaphore, so the finally should be removed.

File: backend/tests/test_streams.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    assert s1 in sm._streams
+    assert f0 != f1
+
+    sm.remove_stream(s0)
+    assert s0 not in sm._streams
+    assert s1 in sm._streams
+
+    sm.remove_stream(s1)
+    assert s0 not in sm._streams
+    assert s1 not in sm._streams
```
Comment (JPHutchins) [inline backend/tests/test_streams.py:94]: Handle the case where the callbacks f0 and f1 are called after the stream has been removed?

Comment (JPHutchins) [reply backend/tests/test_streams.py:94]: Sorta means that every device that has one of those callback functions has to drop it.  That could be as simple as seeing an exception next time it's called.

Comment (JPHutchins) [reply backend/tests/test_streams.py:94]: Nah - the device will own that callback and shouldn't call it if the associated stream has been removed.

**Review by sparklerfish — APPROVED** [review]

**Review by sparklerfish — APPROVED** [review]
LGTM!

**Pull request merged**